mmengine/structures/instance_data.py

```diff
@@ -1,16 +1,28 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import itertools
 from collections.abc import Sized
-from typing import List, Union
+from typing import Any, List, Union
 
 import numpy as np
 import torch
 
+from mmengine.device import get_device
 from .base_data_element import BaseDataElement
 
-IndexType = Union[str, slice, int, list, torch.LongTensor,
-                  torch.cuda.LongTensor, torch.BoolTensor,
-                  torch.cuda.BoolTensor, np.ndarray]
+BoolTypeTensor: Union[Any]
+LongTypeTensor: Union[Any]
+
+if get_device() == 'npu':
+    BoolTypeTensor = Union[torch.BoolTensor, torch.cuda.BoolTensor,
+                           torch.npu.BoolTensor]
+    LongTypeTensor = Union[torch.LongTensor, torch.cuda.LongTensor,
+                           torch.npu.LongTensor]
```

Comment:
Is it necessary to keep `torch.cuda.BoolTensor` when the current_device is 'npu'?

Reply:
Not necessary, already fixed.